### PR TITLE
komga: init at 0.157.0

### PR DESCRIPTION
**Pull request by GovanifY**
###### Description of changes

Komga is a self hosted open source comic book/manga viewer. See https://komga.org .

This packages the released jar as, from what I see, packaging gradle is still a mighty pain.

###### Things done

- Built on platform(s)
  - [X] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [X] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [X] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/servers/komga/default.nix [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+  installPhase = ''
+    mkdir -p "$out/bin"
+    mkdir -p "$out/share/lib/komga/"
+    cp -a ${jar} "$out/share/lib/komga/komga.jar"
+
+    cat <<EOF > "$out/bin/komga"
+    #!/bin/sh
+    ${openjdk11}/bin/java -jar "$out/share/lib/komga/komga.jar"
+    EOF
+    chmod +x "$out/bin/komga"
```
Comment (winterqt) [inline pkgs/servers/komga/default.nix:30]: Prefer `makeWrapper` and `jdk11_headless` here.

```suggestion
    makeWrapper ${jdk11_headless}/bin/java $out/bin/komga --add-flags "-jar $out/share/lib/komga.jar"
```

**Review by winterqt — CHANGES_REQUESTED** [review]

File: pkgs/servers/komga/default.nix [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+  '';
```
[the remark this answers is not in the document]
Comment (GovanifY) [reply pkgs/servers/komga/default.nix]: Agreed out of band that installPhase was the proper way to do this.

File: pkgs/servers/komga/default.nix [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+  nativeBuildInputs = [
+    makeWrapper
+  ];
+
+  installPhase = ''
+    mkdir -p "$out/bin"
+    mkdir -p "$out/share/lib"
+    cp -a "$src" "$out/share/lib/komga.jar"
+    makeWrapper ${jdk11_headless}/bin/java $out/bin/komga --add-flags "-jar $out/share/lib/komga.jar"
+  '';
```
Comment (winterqt) [inline pkgs/servers/komga/default.nix:28]: We can avoid copying the jar here (h/t [`bundletool`](https://github.com/NixOS/nixpkgs/blob/365557e0431c60cba097e678fd7c0ddbb3afd9db/pkgs/development/tools/bundletool/default.nix#L18):

```suggestion
  installPhase = ''
    runHook preInstall
    makeWrapper ${jdk11_headless}/bin/java $out/bin/komga --add-flags "-jar $src"
    runHook postInstall
  '';
```

(Also, you forgot the hooks :)

File: pkgs/servers/komga/default.nix [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+
+  meta = with lib; {
+    description = "Free and open source comics/mangas server";
+    homepage = "https://komga.org/";
+    license = licenses.mit;
+    maintainers = with maintainers; [ govanify ];
+  };
```
Comment (winterqt) [inline pkgs/servers/komga/default.nix:35]: Set `platforms = jdk11_headless.meta.platforms` here.

**Review by winterqt — CHANGES_REQUESTED** [review]

File: pkgs/servers/komga/default.nix [hunk 5]
```diff
@@ -0,0 +1,33 @@
+{ stdenv
+, fetchurl
+, lib
+, makeWrapper
+, jdk11_headless
+}:
+
+stdenv.mkDerivation rec {
```
Comment (winterqt) [inline pkgs/servers/komga/default.nix:8]: Last thing.

```suggestion
stdenvNoCC.mkDerivation rec {
```

**Review by winterqt — CHANGES_REQUESTED** [review]

File: pkgs/servers/komga/default.nix [hunk 6]
```diff
@@ -0,0 +1,33 @@
+{ stdenv
+, fetchurl
+, lib
+, makeWrapper
+, jdk11_headless
+}:
```
Comment (winterqt) [inline pkgs/servers/komga/default.nix:6]: Oh, and this (we prefer `lib` (and `stdenv`, I think) before everything else. I also ordered the rest in order of occurrence in the file, but that really doesn't matter -- feel free to just move `lib`, or reorder the rest as you see fit):

```suggestion
{ lib
, stdenv
, fetchurl
, makeWrapper
, jdk11_headless
}:
```

**Review by winterqt — CHANGES_REQUESTED** [review]

File: pkgs/servers/komga/default.nix [hunk 7]
```diff
@@ -0,0 +1,33 @@
+{ lib
+, stdenvnoCC
+, fetchurl
+, makeWrapper
+, jdk11_headless
+}:
+
+stdenvnoCC.mkDerivation rec {
```
Comment (winterqt) [inline pkgs/servers/komga/default.nix:8]: OfBorg caught this, but not in the way I'd expect (erroring).

```suggestion
stdenvNoCC.mkDerivation rec {
```

**Review by winterqt — CHANGES_REQUESTED** [review]

**Review by winterqt — APPROVED** [review]

**Pull request merged**